api/institutions/views.py

```diff
@@ -0,0 +1,188 @@
+from rest_framework import generics
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from website.models import Institution, Node, User
+
+from api.base import permissions as base_permissions
+from api.base.filters import ODMFilterMixin
+from api.base.views import JSONAPIBaseView
+from api.base.utils import get_object_or_error
+from api.nodes.serializers import NodeSerializer
+from api.users.serializers import UserSerializer
+
+from .serializers import InstitutionSerializer
+
+class InstitutionMixin(object):
+    """Mixin with convenience method get_institution
+    """
+
+    institution_lookup_url_kwarg = 'institution_id'
+
+    def get_institution(self):
+        inst = get_object_or_error(
+            Institution,
+            self.kwargs[self.institution_lookup_url_kwarg],
+            display_name='institution'
+        )
+        return inst
+
+
+class InstitutionList(JSONAPIBaseView, generics.ListAPIView):
+    """
+    Paginated list of verified Institutions affiliated with COS
+
+    ##Institution Attributes
+
+    OSF Institutions have the "institutions" `type`.
+
+        name           type               description
+        -------------------------------------------------------------------------
+        name           string             title of the institution
+        id             string             unique identifier in the OSF
+        logo_path      string             a path to the institution's static logo
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model_class = Institution
+
+    serializer_class = InstitutionSerializer
+    view_category = 'institutions'
+    view_name = 'institution-list'
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        return Institution.find()
+
+
+class InstitutionDetail(JSONAPIBaseView, generics.RetrieveAPIView, InstitutionMixin):
+    """ Details about a given institution.
+
+    ###Permissions
+
+    All institutions are available to be read by everyone. However, no one has write
```

Comment:
I don't think you'll need the bit about write permission. The auto-generated docs will understand this from the RetrieveAPIView class.